expression/builtin_other.go

```diff
@@ -52,6 +61,252 @@ var (
 	_ builtinFunc = &builtinGetParamStringSig{}
 )
 
+type inFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *inFunctionClass) getFunction(ctx context.Context, args []Expression) (sig builtinFunc, err error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	argTps := make([]types.EvalType, len(args))
+	for i := range args {
+		argTps[i] = args[i].GetType().EvalType()
+	}
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETInt, argTps...)
+	bf.tp.Flen = 1
+	switch args[0].GetType().EvalType() {
+	case types.ETInt:
+		sig = &builtinInIntSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InInt)
+	case types.ETString:
+		sig = &builtinInStringSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InString)
+	case types.ETReal:
+		sig = &builtinInRealSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InReal)
+	case types.ETDecimal:
+		sig = &builtinInDecimalSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InDecimal)
+	case types.ETDatetime, types.ETTimestamp:
+		sig = &builtinInTimeSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InTime)
+	case types.ETDuration:
+		sig = &builtinInDurationSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InDuration)
+	case types.ETJson:
+		sig = &builtinInJSONSig{baseBuiltinFunc: bf}
+		sig.setPbCode(tipb.ScalarFuncSig_InJson)
+	}
+	return sig, nil
+}
+
+type builtinInIntSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinInIntSig) evalInt(row types.Row) (int64, bool, error) {
+	sc, args := b.ctx.GetSessionVars().StmtCtx, b.getArgs()
+	arg0, isNull0, err := args[0].EvalInt(row, sc)
+	if isNull0 || err != nil {
+		return 0, isNull0, errors.Trace(err)
+	}
+	isUnsigned0 := mysql.HasUnsignedFlag(args[0].GetType().Flag)
+	var hasNull bool
+	for _, arg := range args[1:] {
+		evaledArg, isNull, err := arg.EvalInt(row, sc)
+		if err != nil {
+			return 0, false, errors.Trace(err)
+		}
+		if isNull {
+			hasNull = true
+			continue
+		}
+		isUnsigned := mysql.HasUnsignedFlag(arg.GetType().Flag)
+		if isUnsigned0 && isUnsigned {
+			if types.CompareUint64(uint64(arg0), uint64(evaledArg)) == 0 {
```

Comment:
`types.CompareUint64()` is a little complicated for this case, we can just check whether `uint64(arg0) == uint64(evaledArg)`